modules/service/src/main/scala/lucuma/odb/service/ObservingModeServices.scala

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.service
+
+import cats.effect.Sync
+import cats.syntax.functorFilter.*
+import cats.syntax.traverse.*
+import edu.gemini.grackle.Result
+import lucuma.core.model.Observation
+import lucuma.odb.data.ObservingModeType
+import lucuma.odb.graphql.input.ObservingModeInput
+import skunk.Session
+import skunk.Transaction
+
+sealed trait ObservingModeServices[F[_]] {
+
+  def createFunction(
+    input: ObservingModeInput.Create
+  ): Result[(List[Observation.Id], Transaction[F]) => F[Unit]]
+
+  def deleteFunction(
+    mode: ObservingModeType
+  ): (List[Observation.Id], Transaction[F]) => F[Unit]
+
+  def updateFunction(
+    input: ObservingModeInput.Edit
+  ): Result[(List[Observation.Id], Transaction[F]) => F[Unit]]
+
+  def createViaUpdateFunction(
+    input: ObservingModeInput.Edit
+  ): Result[(List[Observation.Id], Transaction[F]) => F[Unit]]
+
+}
+
+object ObservingModeServices {
+
+  def fromSession[F[_]: Sync](
+    session: Session[F]
+  ): ObservingModeServices[F] =
+    new ObservingModeServices[F] {
+
+      lazy val gmosLongSlitService: GmosLongSlitService[F] =
+        GmosLongSlitService.fromSession(session)
```

Comment:
Will be filled out with the other modes.  Hopefully this would be the one place to fiddle with when a new mode is added.